CRM/Eventinvitation/Form/Task/ContactSearch.php

```diff
@@ -227,171 +242,194 @@ private function contactsHaveEmails(array $contactIds): bool
-        $emailSender = $senderOptions[$emailSenderId];
-
-        $runnerData = new CRM_Eventinvitation_Object_RunnerData();
-        $runnerData->contactIds = $this->_contactIds;
-        $runnerData->eventId = $values[self::EVENT_ELEMENT_NAME];
-        $runnerData->participantRoleId = $values[self::PARTICIPANT_ROLES_ELEMENT_NAME];
-        if (!empty($values['resource_demand_id'])) {
-            $runnerData->resourceDemandId = $values['resource_demand_id'];
-        }
-        $runnerData->templateId = $values[self::TEMPLATE_ELEMENT_NAME];
+    // evaluate submission
+    $shallBePdfs = isset($values[self::PDFS_INSTEAD_OF_EMAILS_ELEMENT_NAME]);
+    $emailSenderId = $values[self::EMAIL_SENDER_ELEMENT_NAME];
 
-        // Forward back to the search:
-        $targetUrl = html_entity_decode(CRM_Core_Session::singleton()->readUserContext());
+    $senderOptions = $this->getSenderOptions();
+    $emailSender = $senderOptions[$emailSenderId];
 
-        if ($shallBePdfs) {
-            CRM_Eventinvitation_Queue_Runner_Launcher::launchPdfGenerator($runnerData, $targetUrl);
-        } else {
-            CRM_Eventinvitation_Queue_Runner_Launcher::launchEmailSender($runnerData, $emailSender, $targetUrl);
-        }
+    $runnerData = new CRM_Eventinvitation_Object_RunnerData();
+    $runnerData->contactIds = $this->_contactIds;
+    $runnerData->eventId = $values[self::EVENT_ELEMENT_NAME];
+    $runnerData->participantRoleId = $values[self::PARTICIPANT_ROLES_ELEMENT_NAME];
+    if (isset($values['resource_demand_id'])) {
+      $runnerData->resourceDemandId = $values['resource_demand_id'];
     }
+    $runnerData->templateId = $values[self::TEMPLATE_ELEMENT_NAME];
 
-    /**
-     * Get a list of the available/allowed sender email addresses
-     */
-    private function getSenderOptions(): array
-    {
-        $list = [];
-        $query = civicrm_api3(
-            'OptionValue',
-            'get',
-            [
-                'option_group_id' => 'from_email_address',
-                'option.limit' => 0,
-                'return' => 'value,label',
-            ]
-        );
-
-        foreach ($query['values'] as $sender) {
-            $list[$sender['value']] = $sender['label'];
-        }
+    // Forward back to the search:
+    $targetUrl = html_entity_decode(CRM_Core_Session::singleton()->readUserContext());
 
-        return $list;
+    if ($shallBePdfs) {
+      CRM_Eventinvitation_Queue_Runner_Launcher::launchPdfGenerator($runnerData, $targetUrl);
+    }
+    else {
+      CRM_Eventinvitation_Queue_Runner_Launcher::launchEmailSender($runnerData, $emailSender, $targetUrl);
+    }
+  }
+
+  /**
+   * Get a list of the available/allowed sender email addresses.
+   * @return array<string,string>
+   *   'email@example.com' => 'Display Name',
+   * @throws CRM_Core_Exception
+   */
+  private function getSenderOptions(): array {
+
+    /** @var array<string,string> $list */
+    $list = [];
+    $query = (array) civicrm_api3(
```

Comment:
`@phpstan-var array…` instead of a typecast to `array` might be sufficient because `get` API3 calls will always return an array.